### PR TITLE
add alternate ISO/IFL path patterns. Fixes issue #222

### PR DESCRIPTION
**Pull request by rashupee**
Fixes issue #222

### CODE REVIEW

File: gfw/forestchange/api.py [hunk 1]
```diff
@@ -278,12 +278,20 @@ def _classify_request(path):
         rtype = 'latest'
     elif re.match(r'forest-change/%s/admin/ifl/[A-z]{3,3}$' % dataset, path):
         rtype = 'ifl'
-    elif re.match(r'forest-change/%s/admin/ifl/[A-z]{3,3}/\d$' % dataset, path):
-        rtype = 'ifl_id1'        
+    elif re.match(r'forest-change/%s/admin/ifl/[A-z]{3,3}/\d+$' % dataset, path):
+        rtype = 'ifl_id1'
```
Comment (eightysteele) [inline gfw/forestchange/api.py]: style nit: remove trailing whitespace

File: gfw/forestchange/api.py [hunk 2]
```diff
@@ -278,12 +278,20 @@ def _classify_request(path):
...
     elif re.match(r'forest-change/%s/admin/ifl/[A-z]{3,3}$' % dataset, path):
         rtype = 'ifl'
-    elif re.match(r'forest-change/%s/admin/ifl/[A-z]{3,3}/\d$' % dataset, path):
-        rtype = 'ifl_id1'        
+    elif re.match(r'forest-change/%s/admin/ifl/[A-z]{3,3}/\d+$' % dataset, path):
+        rtype = 'ifl_id1'  
+    elif re.match(r'forest-change/%s/ifl/[A-z]{3,3}$' % dataset, path):
+        rtype = 'ifl'
+    elif re.match(r'forest-change/%s/ifl/[A-z]{3,3}/\d+$' % dataset, path):
+        rtype = 'ifl_id1'
```
Comment (eightysteele) [inline gfw/forestchange/api.py]: style nit: remove trailing whitespace

File: test/gfw/forestchange/api_test.py [hunk 3]
```diff
@@ -245,12 +245,30 @@ def test_classify_request(self):
         path = '/forest-change/forma-alerts'
         self.assertEqual(('forma-alerts', 'all'), api._classify_request(path))
 
+        path = '/forest-change/forma-alerts/admin/ifl/bra'
+        self.assertEqual(('forma-alerts', 'ifl'), api._classify_request(path))
+
+        path = '/forest-change/forma-alerts/admin/ifl/bra/123'
+        self.assertEqual(('forma-alerts', 'ifl_id1'), api._classify_request(path))
```
Comment (eightysteele) [inline test/gfw/forestchange/api_test.py]: style nit: line longer than 80 chars

File: test/gfw/forestchange/api_test.py [hunk 4]
```diff
@@ -245,12 +245,30 @@ def test_classify_request(self):
...
+
+        path = '/forest-change/forma-alerts/ifl/bra'
+        self.assertEqual(('forma-alerts', 'ifl'), api._classify_request(path))
+
+        path = '/forest-change/forma-alerts/ifl/bra/123'
+        self.assertEqual(('forma-alerts', 'ifl_id1'), api._classify_request(path))
```
Comment (eightysteele) [inline test/gfw/forestchange/api_test.py]: style nit: line longer than 80 chars

File: test/gfw/forestchange/api_test.py (hunk 3 above)
Comment (rashupee) [inline test/gfw/forestchange/api_test.py]: Hi Aaron, I think I fixed the trailing whitespace and number of chars > 80 problems. :)

Comment (eightysteele) [inline test/gfw/forestchange/api_test.py]: perf :)

File: test/gfw/forestchange/api_test.py (hunk 4 above)
Comment (rashupee) [inline test/gfw/forestchange/api_test.py]: Sorry to not have fixed this earlier. Still getting the hang, of course. I thought the line was 77 characters long - but, of course, with the tab, it's 81... So I split the line, saved, and then ran nosetests which returned OK.

**Comment by apercas** [comment]
Manually merged https://github.com/wri/gfw-api/pull/253

**Pull request closed without merging**